qiskit_aer/circuit/aer_circuit.py

```diff
@@ -0,0 +1,291 @@
+from qiskit.tools.parallel import parallel_map
+from qiskit.providers.options import Options
+from qiskit.qobj import QobjExperimentHeader
+from qiskit_aer.aererror import AerError
+# pylint: disable=import-error, no-name-in-module
+from qiskit_aer.backends.controller_wrappers import AerCircuit_
+
+
+class AerCircuit:
+    """"Aer Internal Circuit class"""
+
+    def __init__(self,
+                 circuit: QuantumCircuit,
+                 shots: int,
+                 seed: Optional[int] = None):
+
+        self._circuit = circuit
+        self._shots = shots
+        self._seed = seed
+
+        self._aer_circ = None
+
+        self._num_qubits = circuit.num_qubits
+        self._num_memory = circuit.num_clbits
+        self._max_conditional_idx = 0
+
+        qreg_sizes = []
+        creg_sizes = []
+        self._global_phase = float(circuit.global_phase)
+
+        for qreg in circuit.qregs:
+            qreg_sizes.append([qreg.name, qreg.size])
+        for creg in circuit.cregs:
+            creg_sizes.append([creg.name, creg.size])
+
+        self._is_conditional_experiment = any(
+            getattr(inst.operation, "condition", None) for inst in circuit.data
+        )
+
+        self._header = QobjExperimentHeader(
+            n_qubits=self._num_qubits,
+            qreg_sizes=qreg_sizes,
+            memory_slots=self._num_memory,
+            creg_sizes=creg_sizes,
+            name=circuit.name,
+            global_phase=self._global_phase,
+        )
+
+        if hasattr(circuit, "metadata") and circuit.metadata:
+            self._header.metadata = copy(circuit.metadata)
+
+        self._qubit_indices = {qubit: idx for idx, qubit in enumerate(circuit.qubits)}
+        self._clbit_indices = {clbit: idx for idx, clbit in enumerate(circuit.clbits)}
+
+    def _append_op(self, aer_circ, inst, conditional_reg_idx):
+        operation = inst.operation
+        qubits = [self._qubit_indices[qubit] for qubit in inst.qubits]
+        clbits = [self._clbit_indices[clbit] for clbit in inst.clbits]
+        name = operation.name
+        label = operation.label
+        params = operation.params if hasattr(operation, "params") else None
+        copied = False
+
+        for i, param in enumerate(params):
+            if (isinstance(param, ParameterExpression) and len(param.parameters) > 0):
+                if not copied:
+                    params = copy(params)
+                    copied = True
+                params[i] = 0.0
+
+        if name in ('ccx', 'ccz', 'cp', 'cswap', 'csx', 'cx', 'cy', 'cz', 'delay', 'ecr',
+                    'h', 'id', 'mcp', 'mcphase', 'mcr', 'mcrx', 'mcry', 'mcrz', 'mcswap',
+                    'mcsx', 'mcu', 'mcu1', 'mcu2', 'mcu3', 'mcx', 'mcx_gray', 'mcy', 'mcz',
+                    'p', 'r', 'rx', 'rxx', 'ry', 'ryy', 'rz', 'rzx', 'rzz', 's', 'sdg', 'swap',
+                    'sx', 'sxdg', 't', 'tdg', 'u', 'x', 'y', 'z', 'u1', 'u2', 'u3',
+                    'cu', 'cu1', 'cu2', 'cu3'):
```

Comment:
A set here will be about ~10x faster because a tuple will need to be iterated over for each lookup:

```suggestion
        if name in {'ccx', 'ccz', 'cp', 'cswap', 'csx', 'cx', 'cy', 'cz', 'delay', 'ecr',
                    'h', 'id', 'mcp', 'mcphase', 'mcr', 'mcrx', 'mcry', 'mcrz', 'mcswap',
                    'mcsx', 'mcu', 'mcu1', 'mcu2', 'mcu3', 'mcx', 'mcx_gray', 'mcy', 'mcz',
                    'p', 'r', 'rx', 'rxx', 'ry', 'ryy', 'rz', 'rzx', 'rzz', 's', 'sdg', 'swap',
                    'sx', 'sxdg', 't', 'tdg', 'u', 'x', 'y', 'z', 'u1', 'u2', 'u3',
                    'cu', 'cu1', 'cu2', 'cu3'}:
```